src/rrt/Tree.hpp

```diff
@@ -23,13 +24,31 @@ namespace RRT {
 template <typename T>
 class Node {
 public:
-    Node(const T& state, Node<T>* parent = nullptr)
-        : _parent(parent), _state(state) {
+    Node(const T& state, Node<T>* parent = nullptr, int dimensions = 2,
+         std::function<void(T, double*)> TToArray = NULL)
+        : _parent(parent), _state(state), _vec(dimensions) {
         if (_parent) {
             _parent->_children.push_back(this);
         }
+        _coordinates = new double[dimensions];
```

Comment:
C++ question: How do I make vec().data() return a `double*` instead of a `const double*`?